### DISCUSSION TITLE
Simplified Chinese localisation

### DISCUSSION THREAD

**Discussion by angelofan** [jaames/iro.js#182] [General]
I have been using this plugin, it is really a very good plugin.

Every time I view the document, I have to use Google Translate, which can be read directly most of the time, but unfortunately, the code will also be translated together, when I need to copy the code, I must manually switch back Displayed in English, and translated again after the copy is completed. But for many Chinese developers, they are not willing to use the translation plug-in for document translation, but more to choose a plugin compatible with Chinese documents.

China Internet of Things development uses this plug-in is a perfect development method. In order to gain influence in this plug-in and capture the better use of Chinese developers, I am willing to translate the docmunt for this plug-in for free.

Is there any way I can contribute my translation for free?

**Comment by angelofan** [comment]
ionic Framework uses [crowdin](https://crowdin.com/) to manage translation

**Comment by jaames** [comment]
@angelofan translation would be great! The documentation on [iro.js.org](https://iro.js.org) is written in markdown here: https://github.com/jaames/iro.js/tree/master/docs

I will work on adding language selection to the iro.js website, but if you want to start translating the documentation into Chinese, my ideal file structure would look like this:

```
docs
├─ foo.md
├─ etc.md
├─ nested
│  └─ foobar.md
└─ zh
   ├─ foo.md
   ├─ etc.md
   └─ nested
      └─ foobar.md
```

> I have to use Google Translate, which can be read directly most of the time, but unfortunately, the code will also be translated together

Ah, I think it's possible to stop Google Translate doing that, I will work on that too :)

**Comment by angelofan** [comment]
@jaames Thank you very much for your answer.

If your English document is updated, the Chinese needs to be marked with the updated location, so that the translator can make other language updates synchronized, such as the [crowdin](https://crowdin.com/) used by the ionic framework I said. This is the first thing you may need. Consider using a platform like [crowdin](https://crowdin.com/) to host translation documents, so that translators in more languages can also join the community translation and contribute to open source software.

At present, if I directly create documents in other languages, if you update the documents in the main language in the future, it will be difficult to compare where changes have been made, unless you need to query the git change records.

iro.js is now influential, and it’s time to make a multi-language compatible step!

How do I proceed? I will wait for your reply....

**Comment by jaames** [comment]
@angelofan I see. Are there any alternatives to crowdin? I agree that it would be helpful to manage translations, but I don't really want to pay a monthly fee for it, especially when the cheapest tier is $16 every month (over half of what iro.js earns in donations)

**Comment by jaames** [comment]
Nevermind! It looks like they have an open-source plan: https://crowdin.com/page/open-source-project-setup-request

I will send a request and see what happens :)

**Comment by angelofan** [comment]
@jaames I found other translation management platforms and hope we can benefit from them.

|name|type|license|link|
|:---|:---:|:---:|:---|
|serge|Free|Private license|https://serge.io/|
|mojito|open source|Apache License 2.0|https://www.mojito.global/|
|smartcat|Contains the $0 plan|-|https://www.smartcat.ai/|
|localise|Need to pay fees|-|https://localise.biz/|
|weblate|Need to pay fees|-|https://weblate.org/|
|crowdin|Need to pay fees|-|https://crowdin.com/|

**Comment by jaames** [comment]
Thanks @angelofan! But no worries, Crowdin just approved my open-source request! 

I'll create files for iro.js there later :)

**Comment by angelofan** [comment]
@jaames It is worth celebrating!

**Comment by jaames** [comment]
Here's a crowdin invite for anyone who wants to help translate: https://crwd.in/irojs :)

**Comment by angelofan** [comment]
@jaames I am translating the document, but I have encountered a problem.

`index.md` cannot be fully translated. Many words in [crowdin](https://crowdin.com/translate/irojs/16/en-zhcn) have become red and are not selectable. When the mouse cursor is hovering in the red area, it will prompt:
```
This string is hidden and should not be translated
```
This may be because a symbol in your Markdown conflicts with crowdin, which incorrectly assumes that you set some words in the document to be hidden and untranslatable.

![image](https://user-images.githubusercontent.com/20255750/84739650-970c7d00-afde-11ea-91ca-0272595def0f.png)

**Comment by jaames** [comment]
@angelofan Oops, sorry about that. I have un-hidden all of the strings now :)

**Comment by angelofan** [comment]
@jaames Still unable to translate, you may need to sync changes to crowdin.

![image](https://user-images.githubusercontent.com/20255750/84755735-8a931f00-aff4-11ea-9e75-dc415fbe4362.png)

**Comment by angelofan** [comment]
@jaames The red part cannot be selected in `index.md`, the other documents are working normally, I have completed all translations, I also checked all the page anchor links, because the target anchor has been translated, the anchor link also needs to be updated, this is easy to go wrong in the future , When other translators update the translated text of the anchor target, all anchor links will not be located.

**Comment by jaames** [comment]
@angelofan okay, I think Ive unhidden them correctly now, maybe try again?

**Comment by angelofan** [comment]
@jaames It may be because of browser caching. Now the translation can be performed normally. At present, I have translated all Chinese. If the documents in the main language are updated in the future, I will continue to make synchronized updates in a timely manner.

**Comment by angelofan** [comment]
@jaames Does the README.MD file in the root directory of the repository need to be translated?

edit: And other Markdown files in the repository, such as `CHANGELOG.md` and `CONTRIBUTING.md`.

**Comment by jaames** [comment]
@angelofan Great! Thank you for your help!

Only files in /docs need to be translated, but I need to add some things to `index.md` for feature text, buttons, etc on the iro.js website :)

**Comment by angelofan** [comment]
@jaames Ok, I suggest that the website can detect the browser's language preference and use it automatically, and there is a drop-down selector to switch languages manually.

**Comment by jaames** [comment]
<img width="863" alt="Screenshot 2020-06-18 at 20 58 19" src="https://user-images.githubusercontent.com/9112876/85066267-87d82b80-b1a6-11ea-919b-b52a768a032c.png">

@angelofan would this font and text size look okay for Chinese audiences?

**Comment by angelofan** [comment]
@jaames No problem at all, it looks good!

By the way: This font is usually used in formal occasions and is very suitable for documents.

**Comment by jaames** [comment]
I still need to figure out how to translate the homepage, but the Chinese translation is live now: https://iro.js.org/zh/guide.html

Thanks again!

**Comment by angelofan** [comment]
https://iro.js.org/

The homepage still needs to be translated, if you need to provide Chinese entries, I will provide them at any time.

**Comment by jaames** [comment]
Thanks! I'm taking a break from working on iro.js for now, but when I am back I will do that :)
